test/ResultTest.cpp

```diff
@@ -0,0 +1,585 @@
+
+  // Test positive case
+  for (auto& generator : getAllSubSplits(idTable)) {
+    Result result{std::move(generator), {0}, LocalVocab{}};
+    result.cacheDuringConsumption(
+        [predictedSize = 0](const std::optional<IdTable>& aggregator,
+                            const IdTable& newTable) mutable {
+          if (aggregator.has_value()) {
+            EXPECT_EQ(aggregator.value().numColumns(), predictedSize);
+          } else {
+            EXPECT_EQ(predictedSize, 0);
+          }
+          predictedSize += newTable.numColumns();
+          return true;
+        },
+        [&](Result aggregatedResult) {
+          EXPECT_TRUE(aggregatedResult.isFullyMaterialized());
+          EXPECT_EQ(aggregatedResult.idTable(), idTable);
+          EXPECT_EQ(aggregatedResult.sortedBy(), std::vector<ColumnIndex>{0});
+        });
+  }
+
+  // Test negative case
+  for (auto& generator : getAllSubSplits(idTable)) {
+    uint32_t callCounter = 0;
+    Result result{std::move(generator), {}, LocalVocab{}};
+    result.cacheDuringConsumption(
+        [&](const std::optional<IdTable>& aggregator, const IdTable&) {
+          EXPECT_FALSE(aggregator.has_value());
+          ++callCounter;
+          return false;
+        },
+        [&](Result) { ++callCounter; });
+    EXPECT_EQ(callCounter, 0);
+  }
+}
+
+// _____________________________________________________________________________
+TEST(Result, verifyApplyLimitOffsetDoesCorrectlyApplyLimitAndOffset) {
+  auto idTable =
+      makeIdTableFromVector({{0, 9}, {1, 8}, {2, 7}, {3, 6}, {4, 5}});
+  LimitOffsetClause limitOffset{2, 2};
+  {
+    uint32_t callCounter = 0;
+    Result result{idTable.clone(), {}, LocalVocab{}};
+    result.applyLimitOffset(
+        limitOffset, [&](std::chrono::microseconds, const IdTable& innerTable) {
+          // NOTE: duration can't be tested here, processors are too fast
+          auto comparisonTable = makeIdTableFromVector({{2, 7}, {3, 6}});
+          EXPECT_EQ(innerTable, comparisonTable);
+          EXPECT_EQ(innerTable.numColumns(), 2);
+          EXPECT_EQ(innerTable.numRows(), 2);
+          ++callCounter;
+        });
+    EXPECT_EQ(callCounter, 1);
+  }
+
+  for (auto& generator : getAllSubSplits(idTable)) {
+    std::vector<size_t> colSizes{};
+    uint32_t totalRows = 0;
+    Result result{std::move(generator), {}, LocalVocab{}};
+    result.applyLimitOffset(
+        limitOffset, [&](std::chrono::microseconds, const IdTable& innerTable) {
+          // NOTE: duration can't be tested here, processors are too fast
+          for (const auto& row : innerTable) {
+            ASSERT_EQ(row.size(), 2);
+            EXPECT_NE(row[0].getVocabIndex().get(), 0);
+            EXPECT_NE(row[0].getVocabIndex().get(), 1);
+            EXPECT_NE(row[0].getVocabIndex().get(), 4);
+            EXPECT_NE(row[1].getVocabIndex().get(), 9);
+            EXPECT_NE(row[1].getVocabIndex().get(), 8);
+            EXPECT_NE(row[1].getVocabIndex().get(), 5);
```

Comment:
Can't this test be more precise or something? Or comment it, I don't quite know what is going on here.